xlive/XLive/UserManagement/CUser.cpp

```diff
@@ -98,51 +112,41 @@ void CUserManagement::CreateUser(const XNADDR* pxna, BOOL user)
 
 		this->secure_map[hostpair] = secure;
 		this->secure_map[hostpair_1000] = secure;
-		this->cusers[secure] = nUser;
-		this->xnmap[secure] = pxna->ina.s_addr;
 
 		this->pmap_a[secure] = nPort_base;
 		this->pmap_b[secure] = nPort_join;
 	}
-	else
-	{
-		/*
-			If the user variable is true, we update cusers "array" to map the secure address to the newly created object...
-			Then map the secure address to the WAN IP (XN->ina) using xnmap, where secure address is the key to the array.
-			Then map the abenet to the secure address via xntosecure[ab], where AB is the key to the array.
-		*/
-		this->cusers[secure] = nUser;
-		this->xnmap[secure] = pxna->ina.s_addr;
-	}
-
-	nUser->bValid = true;
+	this->xnmap[secure] = pxna->ina.s_addr;
 }
 
-// FIXME
 void CUserManagement::UnregisterSecureAddr(const IN_ADDR ina)
 {
 	CUser* to_remove_user = cusers[ina.s_addr];
 	if (to_remove_user != nullptr)
 	{
-		std::pair<ULONG, short> smap, smap1;
-		if (to_remove_user->xnaddr.ina.s_addr == H2Config_ip_wan)
-		{
-			smap = std::make_pair(H2Config_ip_lan, to_remove_user->xnaddr.wPortOnline);
-			smap1 = std::make_pair(H2Config_ip_lan, ntohs(htons(to_remove_user->xnaddr.wPortOnline) + 1));
-		}
-		else
+		if (xnmap[ina.s_addr])
```

Comment:
this also could insert unneeded elements into the array, use `count` or `find`, etc